setup.py

```diff
@@ -48,6 +49,7 @@
         "pyarrow >= 1.0.0, < 4.0dev",
     ],
     "pandas": ["pandas>=0.23.0", "pyarrow >= 1.0.0, < 4.0dev",],
+    "bignumeric_type": ["pyarrow >= 3.0.0, < 4.0dev"],
```

Comment:
I like that we're documenting the supported versions in a machine-checkable way, but I'm also wary of adding extras. Would we consider it a breaking change to remove this in the future when we bump every minimum pyarrow to 3.0?

Reply:
Hmm... I would say not. The existing code itself will still run fine and will not break, perhaps only _installing_ a new library version might break some deployment pipelines when the `bignumeric_type` extra is removed.

The following are the broad categories of BigQuery installations:

- BQ installed without any extra - these should be fine (even if they use BIGNUMERIC, the latter implies that they already have an up-to-date `pyarrow` version that supports it). And fresh installations get the latest supported dependency versions anyway.
- BQ installed with all extras by using the catch-all `"all"` extra - these should be fine, as they already use `pyarrow` 3+ pulled in by `bignumeric_type`. Removing that extra will not downgrade `pyarrow`.
- BQ specifically installed with the `bignumeric_type` extra - installation will emit a warning about the missing extra (at least when using `pip`). This _might_ break a pipeline if it's made sensitive to any warnings, but it's probably not that common a case.

Do you think that they are any other cases that I might have missed?

Reply:
I think you're right that most people will get this via `all`